sdk-core/src/main/java/dev/restate/sdk/core/Entries.java

```diff
@@ -239,18 +253,26 @@ void trace(SleepEntryMessage expected, Span span) {
 
     @Override
     public boolean hasResult(SleepEntryMessage actual) {
-      return actual.hasResult();
+      return actual.getResultCase() != Protocol.SleepEntryMessage.ResultCase.RESULT_NOT_SET;
     }
 
     @Override
     public ReadyResultInternal<Void> parseEntryResult(SleepEntryMessage actual) {
-      return ReadyResults.empty();
+      if (actual.getResultCase() == SleepEntryMessage.ResultCase.FAILURE) {
+        return ReadyResults.failure(Util.toRestateException(actual.getFailure()));
+      } else if (actual.getResultCase() == SleepEntryMessage.ResultCase.EMPTY) {
+        return ReadyResults.empty();
+      } else {
+        throw new IllegalStateException("SleepEntry has not been completed.");
+      }
     }
```

Comment:
I really hope i can improve this stuff at some point by using the protobuf generic apis that let me interact with field numbers directly...